### ISSUE TITLE
cdc: better classify errors when fetching CDC generations

### ISSUE THREAD

**Issue by kbr-** [scylladb/scylla#6804]
This PR: https://github.com/scylladb/scylla/pull/6670
changed the behavior of CDC generation fetching code to stop retrying if an "exotic" error, such as a deserialization error, appears. These errors indicate probable bugs.

However, it doesn't classify all errors properly. For example, the following error message taken from an order version of Scylla:
```
Could not retrieve CDC streams with timestamp 2020/06/18 20:14:37: exceptions::read_failure_exception (Operation failed for system_distributed.cdc_generations - received 1 responses and 2 failures from 2 CL=QUORUM.). Will retry again.
```

would now be classified as an "exotic" error, even though it could happen due to a simple network partition (when the querying node fails to receive responses from a subset of nodes).
This needs to be improved.

**Comment by nyh** [comment]
Already in branch 4.2, and the patch it fixes was not in 4.1. So no need to backport.

**Comment by nyh** [comment]
Sorry, my mistake - actually the fix was *not* in branch 4.2. So backported to 4.2: ad40f9222cac0210231a024bec6998f5d872a9fa
